tensorbay/cli/log.py

```diff
@@ -99,57 +120,87 @@ def _join_branch_names(commit_id: str, branch_names: List[str]) -> str:
     )
 
 
-def _get_oneline_log(commit: Commit, branch_names: Optional[List[str]]) -> str:
+def _get_oneline_commit_log(commit: Commit, branch_names: Optional[List[str]]) -> str:
     commit_id = shorten(commit.commit_id)
     if branch_names:
         commit_id = _join_branch_names(commit_id, branch_names)
-    return _ONELINE_LOG.format(commit_id, commit.title)
+    return _ONELINE_COMMIT_LOG.format(commit_id, commit.title)
+
 
+def _get_oneline_draft_log(draft: Draft) -> str:
+    return _ONELINE_DRAFT_LOG.format(draft.number, draft.title)
 
-def _get_full_log(commit: Commit, branch_names: Optional[List[str]]) -> str:
+
+def _get_full_commit_log(commit: Commit, branch_names: Optional[List[str]]) -> str:
     description = commit.description
     if description:
         description = f"\n\n{indent(description, INDENT)}"
     commit_message = f"{commit.title}{description}\n"
     commit_id = commit.commit_id
     if branch_names:
         commit_id = _join_branch_names(commit_id, branch_names)
-    return _FULL_LOG.format(
+    return _FULL_COMMIT_LOG.format(
         commit_id,
         commit.committer.name,
         datetime.fromtimestamp(commit.committer.date).strftime("%a %b %d %H:%M:%S %y"),
         commit_message,
     )
 
 
-class _CommitPrinter:
-    """This class defines the structure of logging commits.
+def _get_full_draft_log(draft: Draft) -> str:
+    description = draft.description
+    if description:
+        description = f"\n\n{indent(description, INDENT)}"
+    draft_message = f"{draft.title}{description}\n"
+    return _FULL_DRAFT_LOG.format(
+        draft.number,
+        datetime.fromtimestamp(draft.updated_at).strftime("%a %b %d %H:%M:%S %y"),
+        draft_message,
+    )
+
+
+class _Printer:
+    """This class defines the structure of logging commits and open drafts.
 
     Arguments:
         dataset_client: The dataset that needs to be logged.
         revisions: The revisions that needs to be logged.
         oneline: Whether to log with oneline method.
         commit_id_to_branches: The map of commit id to branch name.
+        show_drafts: Whether to log open drafts.
 
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         dataset_client: DatasetClientType,
         revisions: List[Optional[str]],
         commit_id_to_branches: Dict[str, List[str]],
         oneline: bool,
+        show_drafts: bool,
     ):
         all_commits = list(map(dataset_client.list_commits, revisions))
-
-        if len(all_commits[0]) == 0:
-            error(f'Dataset "{dataset_client.name}" has no commit history')
+        all_drafts: List[Draft] = []
+        if show_drafts:
+            for revision in revisions:
+                all_drafts.extend(dataset_client.list_drafts(branch_name=revision))
+
+        if not all_commits[0]:
+            if not all_drafts:
+                error(f'Dataset "{dataset_client.name}" has no commit history or open drafts')
```

Comment:
if `show_drafts` flags is `false`, should the error message contain the info about draft?